LiqoNet: Grafana "Liqo Network" dashboard (#1487)
liqotech/liqo: docs/usage/collect-metrics.md

# Prometheus Metrics

This section presents the metrics exposed by Liqo components, using [Prometheus](https://prometheus.io/) format.

## Scraping metrics

How metrics are scraped will depend on how you're operating your **Prometheus server(s)**.
Liqo presumes the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) is being used to run Prometheus, providing a **ServiceMonitor** resource for each component.

Metrics are **disabled** by default.
In order to allow Prometheus to scrape metrics from Liqo components, you have to include the **liqoctl** `--enable-metrics` flag during installation (checks [custom installation](InstallCustomization) section).
This flag enables the metrics exposition and the **ServiceMonitor** resources creation.

Alternatively it's possible to customize these settings using **helm** (checks the [install with helm](InstallationHelm) section).

## Cross-cluster network metrics

> These metrics are available for each peered remote cluster:

Reviewer: ```suggestion
These metrics are available for each peered remote cluster, providing statistics about the cross-cluster network interconnections:
```

- **liqo_peer_receive_bytes_total**: the total number of bytes received from a remote cluster.
- **liqo_peer_transmit_bytes_total**: the total number of bytes transmitted to a remote cluster.
- **liqo_peer_latency_us**: the latency between the local cluster and a remote cluster.

> - **liqo_peer_is_connected**: a boolean value that indicates if the local cluster is connected to a remote cluster.

Reviewer: ```suggestion
- **liqo_peer_is_connected**: whether the network interconnection is established and works properly.
```

> # Collect metrics

Reviewer: ```suggestion
# Prometheus Metrics
```

Seems clearer to me. I would also rename the file.

> This section presents the metrics exposed by **liqo components**, which can be scraped by [Prometheus](https://prometheus.io/). In order to show how to use those metrics some [Grafana](https://grafana.com/) dashboards are provided.

Reviewer: Be consistent, Liqo is always written uppercase in the documentation.

## Prerequisites

In order to collect metrics from Liqo components, you need to install Prometheus and Grafana.

> It's suggested to install the **kube-prometheus** stack to deploy them. Instructions can be found [here](https://github.com/prometheus-operator/kube-prometheus)

Reviewer: I wouldn't say you have to install Prometheus, as if you care about observability it is likely you already have it installed. You could rephrase saying that you assume it to be present, similarly to what is done [here](https://cert-manager.io/docs/usage/prometheus-metrics/) for instance.

> ## How to enable metrics

Reviewer: ```suggestion
## Scraping metrics
```

> Metrics are **disabled** by default. In order to allow Prometheus to scrape metrics from liqo components, you have to enable metrics by setting the `metrics.enabled` value to `true` in the `values.yaml` file. To enable the creation of **ServiceMonitor** resources for Prometheus, you can set the `metrics.serviceMonitor.enabled` value to `true`.

Reviewer: * Please, stick to one sentence per line, in order to limit the issues with git versioning.
* The standard way to enable metrics collection is to use the appropriate liqoctl flag, but you did not mention that here. You could also mention that advanced aspects can be customized through the helm values file.

> ## Network

Reviewer: ```suggestion
## Cross-cluster network metrics
```

Just a suggestion though.

> Those metrics are related to the peering with other clusters. Those are:

Reviewer: The title mentions networking, while here you talk about peering. This looks confusing to me, and I would suggest to rephrase the sentence a bit.

- **liqo_peer_receive_bytes_total**: the total number of bytes received from a remote cluster.
- **liqo_peer_transmit_bytes_total**: the total number of bytes transmitted to a remote cluster.
- **liqo_peer_latency_us**: the latency between the local cluster and a remote cluster.
- **liqo_peer_is_connected**: a boolean value that indicates if the local cluster is connected to a remote cluster.

### Grafana dashboard

We provide a grafana dashboard to monitor the network status.\
You can download the dashboard {download}`here</_downloads/grafana/liqonetwork.json>`.

> A screenshot of the dashboard is shown below.

Reviewer: I would personally suggest to elaborate a bit more. For instance:

```
We provide a {download}`sample Grafana dashboard </_downloads/grafana/liqonetwork.json>` to monitor the network interconnection of an arbitrary number of Liqo peerings.
As presented in the screenshot below, it includes an overview section presenting the overall cross-cluster throughput, followed by detailed per-peering throughput and latency information.
```
